Makefile

```diff
@@ -62,3 +71,18 @@ build: clean
 .PHONY: clean
 clean:
 	rm -rf dist
+
+
+# full clean
+.PHONY: full-clean
+full-clean:: local-uninstall clean
+	rm -vrf *.egg-info || : ;
+	rm -vfrd ./piplicenses/__pycache__ 2>/dev/null || : ;
+	rm -vfrd ./.coverage 2>/dev/null || : ;
+	rm -vfrd ./.mypy_cache 2>/dev/null || : ;
+	rm -vfrd ./.pytest_cache 2>/dev/null || : ;
+
+# hard reset
+.PHONY: un-setup
+un-setup:: full-clean
+	rm -vfrd ./venv 2>/dev/null || : ;
```

Comment:
This removes more than the actually used virtual environment, possibly removing user-specific directories (I sometimes have a second virtual environment in there which uses other dependency versions or another Python interpreter version).

Reply:
This is intended as an inverse of the venv and setup target to ensure reversible changes. This is a separate target for niche cases like yours. I use this automation to keep my environment pristine and ready for fresh development. I loath debugging edge-cases that arise from non-idempotent automation. 

Consider the use of `tox` for cross-python testing in virtual environments, as is best-practice.

I also strongly suggest any worthwhile nested venv testing should be automated as well (in the same spirit of consistency)